### ISSUE TITLE
Focus ring missing on sidebar's vertical toolbar after moving elements into shadow root

### ISSUE THREAD

**Issue by robertknight** [hypothesis/client#2959]
We recently moved all of the DOM elements in the sidebar's vertical toolbar into a shadow root. An overlooked side effect of that was that buttons in the sidebar's vertical toolbar no longer show a focus ring when these buttons are focused using the keyboard.

It looks to me like this is an issue with our polyfill for the `:focus-visible` pseudo-class not working because it relies on a `js-focus-visible` class that is applied to the document body.

**Steps to reproduce:**

1. Go to https://hypothes.is/docs/help
2. Focus the toggle button for the sidebar using the keyboard.

**Expected result:**

A visible focus ring should be shown on the button.

**Actual result:**

Testing in Chrome 90 (dev), Firefox 85 and Safari 14, no focus ring is visible:

<img width="88" alt="Sidebar toolbar focused" src="https://user-images.githubusercontent.com/2458/106920373-eedd2d00-6702-11eb-856e-3ebb1b5115da.png">

See also https://github.com/hypothesis/client/issues/2689.

**Comment by esanzgar** [comment]
https://github.com/WICG/focus-visible#shadow-dom
